src/os/bluestore/KernelDevice.cc

```diff
@@ -525,6 +525,60 @@ void KernelDevice::aio_submit(IOContext *ioc)
   }
 }
 
+int KernelDevice::write(
```

Comment:
I'd suggest to reuse the sync path from aio_write code as much as possible here.... may be make a standalone internal func?